pkg/apis/serving/v1alpha1/kfservice_status.go

```diff
@@ -14,41 +14,82 @@ limitations under the License.
 package v1alpha1
 
 import (
-	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"github.com/knative/pkg/apis"
 	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"k8s.io/api/core/v1"
 )
 
 // ConditionType represents a Service condition value
 const (
-	// ServiceConditionRoutesReady is set when the service's underlying
+	// KFServiceConditionRoutesReady is set when the service's underlying
 	// routes have reported readiness.
-	ServiceConditionRoutesReady duckv1alpha1.ConditionType = "RoutesReady"
-	// ServiceConditionDefaultConfigurationsReady is set when the service's underlying
+	KFServiceConditionRoutesReady apis.ConditionType = "RoutesReady"
+	// KFServiceConditionDefaultConfigurationsReady is set when the service's underlying
 	// default configuration have reported readiness.
-	ServiceConditionDefaultConfigurationsReady duckv1alpha1.ConditionType = "DefaultConfigurationReady"
+	KFServiceConditionDefaultConfigurationsReady apis.ConditionType = "DefaultConfigurationReady"
 	// ServiceConditionCanaryConfigurationsReady is set when the service's underlying
 	// canary configuration have reported readiness.
-	ServiceConditionCanaryConfigurationsReady duckv1alpha1.ConditionType = "CanaryConfigurationReady"
+	KFServiceConditionCanaryConfigurationsReady apis.ConditionType = "CanaryConfigurationReady"
 )
 
-var serviceCondSet = duckv1alpha1.NewLivingConditionSet(
-	ServiceConditionDefaultConfigurationsReady,
-	ServiceConditionCanaryConfigurationsReady,
-	ServiceConditionRoutesReady,
+// KFService Ready condition is depending on default configuration and route readiness condition
+// canary configuration readiness condition only present when canary is used and currently does
+// not affect KFService readiness condition.
+var serviceCondSet = apis.NewLivingConditionSet(
+	KFServiceConditionDefaultConfigurationsReady,
+	KFServiceConditionRoutesReady,
 )
 
+var _ apis.ConditionsAccessor = (*KFServiceStatus)(nil)
+
+func (ss *KFServiceStatus) InitializeConditions() {
+	serviceCondSet.Manage(ss).InitializeConditions()
+}
+
+// IsReady returns if the service is ready to serve the requested configuration.
+func (ss *KFServiceStatus) IsReady() bool {
+	return serviceCondSet.Manage(ss).IsHappy()
+}
+
+// GetCondition returns the condition by name.
+func (ss *KFServiceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return serviceCondSet.Manage(ss).GetCondition(t)
+}
+
 // PropagateDefaultConfigurationStatus propagates the default Configuration status and applies its values
 // to the Service status.
 func (ss *KFServiceStatus) PropagateDefaultConfigurationStatus(dcs *knservingv1alpha1.ConfigurationStatus) {
 	ss.Default.Name = dcs.LatestCreatedRevisionName
-	//TODO @yuzisun populate configuration status conditions
+	cc := dcs.GetCondition(knservingv1alpha1.ConfigurationConditionReady)
```

Comment:
I'm finding this variable naming of cc, dcs, etc to be really confusing to read. Are you married to it?